src/java/org/apache/cassandra/index/sai/disk/format/Version.java

```diff
@@ -103,6 +105,47 @@ public static Version latest()
         return LATEST;
     }
 
+    /**
+     * Calculates the maximum allowed length for SAI index names to ensure generated filenames
+     * do not exceed the system's filename length limit (defined in SchemaConstants.FILENAME_LENGTH).
+     * This accounts for all additional components in the filename.
+     */
+    public static int calculateIndexNameAllowedLength()
+    {
+
+        int addedLength = getAddedLengthFromDescriptorAndVersion();
+        assert addedLength < SchemaConstants.FILENAME_LENGTH;
+        return SchemaConstants.FILENAME_LENGTH - addedLength;
+    }
+
+    /**
+     * Calculates the length of the added prefixes and suffixes from Descriptor constructor
+     * and Version.stargazerFileNameFormat.
```

Comment:
Nit: A link might make things easier for IDEs if the referenced method changes in the future:
```suggestion
     * and {@link Version#stargazerFileNameFormat}.
```